File: R/combinations.R
.packageName <- "bise"

.First.lib <- function(lib, pkg) {
              library.dynam("bise", pkg, lib)
              print("test version of bise loaded")
}


combinations <- function(id) {
    n <- length(id)
    ncomb <- (n^2-n) / 2
    stat1 <- vector(mode="numeric", length=ncomb)
    stat2 <- vector(mode="numeric", length=ncomb)
    res <- .Fortran("combinations", id=as.integer(id), n=as.integer(n), ncomb=as.integer(ncomb), stat1=as.integer(stat1), stat2=as.integer(stat2), PACKAGE="bise")
    attach(res)
    return(c(stat1,stat2))
}

File: src/combinations.f
      subroutine combinations(id,n,ncomb,stat1,stat2)
      
      implicit none
      
      integer id(*),n,ncomb,x,y,count
      integer stat1(*),stat2(*)
      
      count = 0
      
      do x=1,n-1
      	 do y=x+1,n
            count = count + 1
            stat1(count) = id(x)
            stat2(count) = id(y)
         enddo
      enddo
      
      return
      end
